From ec007e65ae5a8e7ceef32b9d342ecfe998ab3fae Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Mon, 29 Dec 2008 17:07:44 +0800
Subject: [PATCH 034/280] ENGR00103770 vpu: fix chromaInterleave flag set to
 wrong bit on MX51

Fix chromaInterleave flag set to wrong bit on i.MX51.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index b0487bf..fd14797 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -1534,9 +1534,13 @@ RetCode vpu_DecOpen(DecHandle * pHandle, DecOpenParam * pop)
 	/* Clear stream end flag */
 	VpuWriteReg(BIT_BIT_STREAM_PARAM,
 		    VpuReadReg(BIT_BIT_STREAM_PARAM) & ~(1 << (instIdx + 2)));
+
+#ifdef  IMX37_3STACK
 	VpuWriteReg(BIT_FRAME_MEM_CTRL,
 			    VpuReadReg(BIT_FRAME_MEM_CTRL) |
 				(pDecInfo->openParam.chromaInterleave << (instIdx + 1)));
+#endif
+
 	UnlockVpu(vpu_semap);
 
 	return RETCODE_SUCCESS;
-- 
1.8.0

